coverage.txt

```diff
@@ -1,50 +1,52 @@
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/mockData             |      100 |      100 |      100 |      100 |                   |
-  matchMedia.mock.js      |      100 |      100 |      100 |      100 |                   |
-  youtube-videos-mock.js  |      100 |      100 |      100 |      100 |                   |
- src/pages/Home           |      100 |      100 |      100 |      100 |                   |
-  Home.page.jsx           |      100 |      100 |      100 |      100 |                   |
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/pages/Login          |        0 |      100 |        0 |        0 |                   |
-  Login.page.jsx          |        0 |      100 |        0 |        0 |   8,9,12,13,14,17 |
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/pages/NotFound       |        0 |      100 |        0 |        0 |                   |
-  NotFound.page.jsx       |        0 |      100 |        0 |        0 |                 7 |
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/pages/VideoDetail    |    94.44 |       50 |       80 |      100 |                   |
-  VideoDetail.page.jsx    |    91.67 |       50 |       80 |      100 |                22 |
-  VideoDetail.styles.jsx  |      100 |      100 |      100 |      100 |                   |
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/providers/Auth       |    52.94 |        0 |       40 |    52.94 |                   |
-  Auth.provider.jsx       |    52.94 |        0 |       40 |    52.94 |... 13,27,28,32,33 |
-  index.js                |        0 |        0 |        0 |        0 |                   |
- src/utils                |    71.43 |      100 |    33.33 |    71.43 |                   |
-  api.js                  |       50 |      100 |        0 |       50 |         2,9,22,34 |
-  constants.js            |      100 |      100 |      100 |      100 |                   |
-  fns.js                  |    91.67 |      100 |    66.67 |    91.67 |                 2 |
-  storage.js              |    57.14 |      100 |       50 |    57.14 |            7,8,13 |
- src/utils/hooks          |        0 |        0 |        0 |        0 |                   |
-  useVideos.js            |        0 |        0 |        0 |        0 |... 23,24,25,28,31 |
---------------------------|----------|----------|----------|----------|-------------------|
+--------------------------|---------|----------|---------|---------|-------------------
+File                      | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
+--------------------------|---------|----------|---------|---------|-------------------
+All files                 |   67.31 |       25 |   51.11 |   67.11 |                   
```

Comment:
The branch percentage is a bit low, this is a bit tricky to increase since you have to add multiple cases to the tests, when you have `if` statements you have to test the `if` and when there is an `else` you have to test the `else` also, the branch is the lowest percentage in most of the challenges I have reviewed